utilities/roswtf/src/roswtf/__init__.py

```diff
@@ -91,33 +91,42 @@ def roswtf_main():
         print(str(e), file=sys.stderr)
         
 def _roswtf_main():
-    launch_files = names = None
     # performance optimization
     rospack = rospkg.RosPack()
     all_pkgs = rospack.list()
 
-    import optparse
-    parser = optparse.OptionParser(usage="usage: roswtf [launch file]", description="roswtf is a tool for verifying a ROS installation and running system. Checks provided launchfile if provided, else current stack or package.")
+    import argparse
+    parser = argparse.ArgumentParser(usage="usage: roswtf [launch file]", description="roswtf is a tool for verifying a ROS installation and running system. Checks provided launchfile if provided, else current stack or package.")
     # #2268
-    parser.add_option("--all", 
+    parser.add_argument("--all", 
                       dest="all_packages", default=False,
                       action="store_true",
                       help="run roswtf against all packages")
     # #2270
-    parser.add_option("--no-plugins", 
+    parser.add_argument("--no-plugins", 
                       dest="disable_plugins", default=False,
                       action="store_true",
                       help="disable roswtf plugins")
 
-    parser.add_option("--offline", 
+    parser.add_argument("--offline", 
                       dest="offline", default=False,
                       action="store_true",
                       help="only run offline tests")
 
+    parser.add_argument("launch_files",  nargs='*',
+                        help="launch files to check. 0...n files can be passed")
     #TODO: --all-pkgs option
-    options, args = parser.parse_args()
-    if args:
-        launch_files = args
+    args = parser.parse_args()
+    roswtf_node(args)
+
+
+def roswtf_node(args):
```

Comment:
Why should this be a public API? A caller will need to know internal knowledge (what arguments need to be set on the passed `args`) which makes it hard to use. Therefore I think this should either be private or expose a signature with specific keyword arguments.

Reply:
@130s Friendly ping.

Reply:
> Therefore I think this should either be private or expose a signature with specific keyword arguments.

Thanks, I did the latter of your suggestion.